src/main/java/org/opensearch/knn/index/KNNMethodContext.java

```diff
@@ -50,19 +52,18 @@ public class KNNMethodContext implements ToXContentFragment, Writeable {
 
     public static synchronized KNNMethodContext getDefault() {
         if (defaultInstance == null) {
-            defaultInstance = new KNNMethodContext(
-                KNNEngine.DEFAULT,
-                SpaceType.DEFAULT,
-                new MethodComponentContext(METHOD_HNSW, Collections.emptyMap())
-            );
+            MethodComponentContext methodComponentContext = new MethodComponentContext(METHOD_HNSW, Collections.emptyMap());
+            methodComponentContext.setIndexVersion(Version.CURRENT);
```

Comment:
can we add the same thing in the comment too. Otherwise it is really hard to understand why we did this. I would really like to think how we can avoid this.

Reply:
What I mean is this `getDefault` method is used in unit test only. The reason that it worked without indexVersion before was there was no MethodFieldMapperTests. Let me just move this entire method inside test package....